2022-10-27 12:32:58 info: ---------------------------------------
2022-10-27 12:32:58 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 12:32:58 info: ---------------------------------------
2022-10-27 12:33:06 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✓ should call `new Promise()`
    ✓ should take a single argument (1 ms)
    ✓ `resolve()` should be called with a one argument
    ✕ `reject()` should be called with a one argument (2 ms)
    ✓ should resolve when called with a string argument (1 ms)
    ✓ should reject with an Error object when called without an argument

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● getAnonName › `reject()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      66 |
      67 |   test('`reject()` should be called with a one argument', () => {
    > 68 |     expect(state.reject).toBe(1);
         |                          ^
      69 |   });
      70 |
      71 |   test('should resolve when called with a string argument', async () => {

      at Object.toBe (3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js:68:26)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 7 passed, 9 total
Snapshots:   0 total
Time:        1.604 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-10-27 12:33:12 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  29:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2022-10-27 12:33:16 error: *** Spell Checker Report ***

ex1-johnWho.js:13:34 - Unknown word (rejecte)
ex1-johnWho.js:15:7 - Unknown word (rejecte)

2022-10-27 12:46:23 info: ---------------------------------------
2022-10-27 12:46:23 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 12:46:23 info: ---------------------------------------
2022-10-27 12:46:31 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✓ should call `new Promise()`
    ✓ should take a single argument
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✓ should resolve when called with a string argument (2 ms)
    ✓ should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        1.374 s, estimated 2 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-10-27 12:46:37 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  29:1  warning  Remove commented-out code  hyf/no-commented-out-code
  31:1  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2022-10-27 12:54:03 info: ---------------------------------------
2022-10-27 12:54:03 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 12:54:03 info: ---------------------------------------
2022-10-27 12:54:10 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✓ should call `new Promise()`
    ✓ should take a single argument
    ✓ `resolve()` should be called with a one argument (1 ms)
    ✓ `reject()` should be called with a one argument
    ✓ should resolve when called with a string argument (2 ms)
    ✓ should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        1.392 s, estimated 2 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-10-27 12:55:11 info: ---------------------------------------
2022-10-27 12:55:11 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 12:55:11 info: ---------------------------------------
2022-10-27 12:55:16 info: All unit tests passed.
2022-10-27 12:55:23 info: All steps were completed successfully
2022-10-27 12:55:51 info: ---------------------------------------
2022-10-27 12:55:51 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 12:55:51 info: ---------------------------------------
2022-10-27 12:55:53 info: All unit tests passed.
2022-10-27 12:55:56 info: All steps were completed successfully
2022-10-27 01:18:07 info: ---------------------------------------
2022-10-27 01:18:07 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-27 01:18:07 info: ---------------------------------------
2022-10-27 01:18:14 info: All unit tests passed.
2022-10-27 01:18:23 info: All steps were completed successfully
2022-10-27 01:42:27 info: -------------------------------------------------
2022-10-27 01:42:27 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-27 01:42:27 info: -------------------------------------------------
2022-10-27 01:42:35 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✓ should call new Promise()
    ✓ `resolve()` should be called with a one argument (1 ms)
    ✓ `reject()` should be called with a one argument
    ✓ should be a function that takes a single argument
    ✓ (9) should return a rejected promise with an Error object (1 ms)
    ✕ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✓ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 10]

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 99]

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 8 passed, 11 total
Snapshots:   0 total
Time:        1.446 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-10-27 01:46:39 info: -------------------------------------------------
2022-10-27 01:46:39 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-27 01:46:39 info: -------------------------------------------------
2022-10-27 01:46:46 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (1 ms)
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✓ should call new Promise()
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should be a function that takes a single argument
    ✓ (9) should return a rejected promise with an Error object (1 ms)
    ✕ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✓ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 10]

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 99]

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 8 passed, 11 total
Snapshots:   0 total
Time:        1.437 s, estimated 2 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-10-27 01:49:43 info: -------------------------------------------------
2022-10-27 01:49:43 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-27 01:49:43 info: -------------------------------------------------
2022-10-27 01:49:51 info: All unit tests passed.
2022-10-27 01:50:01 info: All steps were completed successfully
2022-10-27 02:00:36 info: -------------------------------------------------
2022-10-27 02:00:36 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-27 02:00:36 info: -------------------------------------------------
2022-10-27 02:00:43 info: All unit tests passed.
2022-10-27 02:00:52 info: All steps were completed successfully
2022-10-27 10:05:39 info: ---------------------------------------
2022-10-27 10:05:39 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-27 10:05:39 info: ---------------------------------------
2022-10-27 10:05:47 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (1 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ should call `new Promise()`
    ✕ `resolve()` should be called with a one argument (2 ms)
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when the die settles successfully (2 ms)
    ✓ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● rollDie › `resolve()` should be called with a one argument

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      42 |
      43 |   test('`resolve()` should be called with a one argument', () => {
    > 44 |     expect(state.resolve).toBe(1);
         |                           ^
      45 |   });
      46 |
      47 |   test('`reject()` should be called with a one argument', () => {

      at Object.toBe (3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js:44:27)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        1.424 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-10-27 10:06:46 info: ---------------------------------------
2022-10-27 10:06:46 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-27 10:06:46 info: ---------------------------------------
2022-10-27 10:06:51 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when the die settles successfully (2 ms)
    ✓ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        1.287 s, estimated 2 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-10-27 10:07:40 info: ---------------------------------------
2022-10-27 10:07:40 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-27 10:07:40 info: ---------------------------------------
2022-10-27 10:07:44 info: All unit tests passed.
2022-10-27 10:07:48 info: All steps were completed successfully
2022-10-27 11:04:26 info: --------------------------------------------
2022-10-27 11:04:26 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-10-27 11:04:26 info: --------------------------------------------
2022-10-27 11:04:34 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-pokerDiceAll.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js
  ex4-pokerDiceAll
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (1 ms)
    ✓ `rollDice` should not contain unneeded console.log calls
    ✓ should use `dice.map()`
    ✓ should use `Promise.all()` (1 ms)
    ✕ should resolve when all dice settle successfully (3 ms)
    ✕ should reject with an Error when a die rolls off the table (4 ms)

  ● ex4-pokerDiceAll › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● ex4-pokerDiceAll › should resolve when all dice settle successfully

    RangeError: Maximum call stack size exceeded

      29 | function rollDice() {
      30 |   // TODO Refactor this function
    > 31 |   const dice = [1, 2, 3, 4, 5];
         |                ^
      32 |
      33 |   const result = Promise.all(dice.map((die) => rollDice(die)));
      34 |

      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:31:16)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)
      at rollDice (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:48)
          at Array.map (<anonymous>)
      at map (3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:33:35)

  ● ex4-pokerDiceAll › should resolve when all dice settle successfully

    expect.assertions(4)

    Expected four assertions to be called but received one assertion call.

      46 |
      47 |   test('should resolve when all dice settle successfully', async () => {
    > 48 |     expect.assertions(4);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:48:12)

  ● ex4-pokerDiceAll › should reject with an Error when a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      63 |
      64 |   test('should reject with an Error when a die rolls off the table', async () => {
    > 65 |     expect.assertions(3);
         |            ^
      66 |     expect(exported).toBeDefined();
      67 |
      68 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex4-pokerDiceAll.test.js:65:12)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 4 passed, 7 total
Snapshots:   0 total
Time:        1.572 s
Ran all test suites matching /ex4-pokerDiceAll.test.js/i.
2022-10-27 11:04:40 error: *** ESLint Report ***

ex4-pokerDiceAll.js
  27:7  error  'rollDie' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)


2022-10-27 11:12:38 info: --------------------------------------------
2022-10-27 11:12:38 info: >>> Running Unit Test `ex4-pokerDiceAll` <<<
2022-10-27 11:12:38 info: --------------------------------------------
2022-10-27 11:12:44 info: All unit tests passed.
2022-10-27 11:12:51 info: All steps were completed successfully
2022-10-27 11:33:54 info: ----------------------------------------------
2022-10-27 11:33:54 info: >>> Running Unit Test `ex5-pokerDiceChain` <<<
2022-10-27 11:33:54 info: ----------------------------------------------
2022-10-27 11:34:01 info: All unit tests passed.
2022-10-27 11:34:11 info: All steps were completed successfully
